### PR TITLE
Add an example based on git history of the project

### PR DESCRIPTION
**Pull request by mcnuttandrew**
This PR adds an interesting looking graphic to our examples:

screen shot 2017-07-31 at 5 34 31 pm

The data is based on the commit history of across react-vis's history. In the near future we should probably shuffle the /charts page on the docs site so that we are showing more than just the showcases, but that's a story for another day.

### CODE REVIEW

File: showcase/examples/history/history-example.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  return {
+    ...lastCommit,
+    label: lastCommit.commiter.toUpperCase(),
+    style: {
+      fontSize: '8px'
+    }
+  };
+});
+
+const FlexibleXYPlot = makeWidthFlexible(XYPlot);
```
Comment (Apercu) [inline showcase/examples/history/history-example.js]: I saw we are doing this quite a lot, what do you think about exporting the three different types of Flexible plots directly for easier usability?

File: showcase/examples/history/history-example.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  );
+});
+
+const authors = commitsByAuthor.map(commits => {
+  const lastCommit = commits[commits.length - 1];
+  return {
+    ...lastCommit,
+    label: lastCommit.commiter.toUpperCase(),
+    style: {
+      fontSize: '8px'
```
Comment (Apercu) [inline showcase/examples/history/history-example.js]: I personally prefer to use the numeric representation when dealing with pixels, but nit

File: showcase/examples/history/history-example.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+      <div className="history-example">
+        <div className="chart">
+          <FlexibleXYPlot
+            animation
+            margin={{top: 50, bottom: 50, right: 50, left: 50}}
+            yType="log"
+            xType="time"
+            height={500}>
+            <XAxis />
+            <YAxis tickFormat={t => t}/>
```
Comment (Apercu) [inline showcase/examples/history/history-example.js]: should we provide the identity function as the default `tickFormat`?

File: showcase/examples/history/history-extractor-script.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+const fs = require('fs');
+const util = require('util');
+// only runs on node 8 so be careful!
+const exec = util.promisify(require('child_process').exec);
+
+async function buildHistory() {
```
Comment (Apercu) [inline showcase/examples/history/history-extractor-script.js]: wew fancy

File: showcase/examples/history/history-extractor-script.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+const fs = require('fs');
+const util = require('util');
+// only runs on node 8 so be careful!
```
Comment (Apercu) [inline showcase/examples/history/history-extractor-script.js]: is the error message explicit about it? If not might make sense to check for the running node version. `process.versions.node` if I'm not mistaken

File: showcase/examples/history/history-example.js (hunk 1 above)
Comment (mcnuttandrew) [inline showcase/examples/history/history-example.js]: I'm for that, but that's a maybe different PR? I'd bet @jckr has some opinions about that

File: showcase/examples/history/history-example.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+const FlexibleXYPlot = makeWidthFlexible(XYPlot);
+
+export default class HistoryExample extends React.Component {
+  render() {
+    return (
+      <div className="history-example">
+        <div className="chart">
+          <FlexibleXYPlot
+            animation
+            margin={{top: 50, bottom: 50, right: 50, left: 50}}
```
Comment (mcnuttandrew) [inline showcase/examples/history/history-example.js]: wow! i did not know about that, til

File: showcase/examples/history/history-example.js (hunk 3 above)
Comment (mcnuttandrew) [inline showcase/examples/history/history-example.js]: I kinda like the smart-ish defaults that we have in there now? It allows XYPlot to offers its knowledge about the situation pretty well

File: showcase/examples/history/history-example.js (hunk 1 above)
Comment (Apercu) [inline showcase/examples/history/history-example.js]: Yes sure, was just saying before forgetting ;p

File: showcase/examples/history/history-example.js (hunk 3 above)
Comment (Apercu) [inline showcase/examples/history/history-example.js]: okay

File: showcase/examples/history/history-example.js (hunk 1 above)
Comment (mcnuttandrew) [inline showcase/examples/history/history-example.js]: file an issue?

**Pull request merged**